refactor: use Manager to filter comments

Woser-tan opened the pull request:
Used managers to collect the approved comments of a Post model.

Woser-tan committed:
Tried filtering with models.Manager

Woser-tan committed:
Filtered approved comments usingManagers

Woser-tan committed:
Filtered approved comments with Managers

gurolcay commented:
.idea klasorunu gitignore'a ekleyelim

projelerinde gitignore.io'dan yararlanabilirsin

gurolcay commented on blog/migrations/0003_alter_comment_managers.py:
```diff
@@ -0,0 +1,20 @@
+import django.db.models.manager
+
+
+class Migration(migrations.Migration):
```
tek migration'a indirelim. 
hint: pythonda migration geri alma var. belli bir yere migration geri alip (0002'inci arkadasa geri almalisin), sonrasinda model uzerinde nihai degisikligini yaptiginda migration olusturmalisin.

gurolcay commented on blog/models.py:
```diff
@@ -4,34 +4,48 @@
 
 # Create your models here.
+class Comments(models.Manager):
+    def get_all(self):
```
`get_all` anlamsiz olmus, bunu kaldirabiliriz.

gurolcay commented on blog/models.py:
```diff
@@ -4,34 +4,48 @@
+    def get_all(self):
+        return self.comments.all()
+    def get_approved(self):
+        return self.comments.filter(approved_comment=True)
```
burada self zaten kendisi olacak, `self.comments` olmali.

`get_approved` -> `approved`

gurolcay commented on blog/models.py:
```diff
@@ -4,34 +4,48 @@
-        return self.comments.filter(approved_comment=True)
-    
+
+        return Comments.get_approved(self)
```
bu kullanim yanlis olmus burada metot dursun.

`self.comments.approved()`

gurolcay commented on blog/urls.py:
```diff
@@ -3,14 +3,24 @@
-]
\ No newline at end of file
+    path('', views.PostView.as_view(view_select='post_list'), name = 'post_list'),
+    #path('', views.PostList.as_view(), name = 'post_list'),
```
yorum satirlarini ucuralim

gurolcay commented on blog/views.py:
```diff
@@ -6,79 +6,198 @@
+
+        posts = Post.objects.filter(published_date__lte = timezone.now()).order_by('published_date')
+
+        if(view_select == 'post_list'):
```
`view_select` silinsin burada PostListView... seklinde olsun. DRF entegre ettigimizde buraya geri donecegiz. Tek view orada olacak.

gurolcay commented on blog/views.py:
```diff
@@ -101,97 +221,79 @@ def get(self, request, pk):
+"""    
+
+"""
+def post_list(request):
```
bunlari silebilriz artik

gurolcay commented:
MR title daha anlanli olmali. Bu is kapsaminda biz ne yaptik neden yaptik gibi sorulara cevap bulabilmeliyim.

Woser-tan committed:
refactor(Views, URLs): adhere to proper design pattern.

Reverted to individual class-based design for Views. Refactored URL's accordingly. Refactored the use of Manager's to get approved comments of a Post instance.

Woser-tan committed:
docs: use gitignore.io

Woser-tan committed:
Delete .idea directory
